File: src/main/java/seremis/geninfusion/soul/allele/AlleleIntArray.java
package seremis.geninfusion.soul.allele;

import net.minecraft.nbt.NBTTagCompound;
import seremis.geninfusion.api.soul.EnumAlleleType;
import seremis.geninfusion.soul.Allele;

public class AlleleIntArray extends Allele {

    public int[] value;

    public AlleleIntArray(boolean isDominant, int[] value) {
        super(isDominant, EnumAlleleType.INT_ARRAY);
        this.value = value;
    }

    public AlleleIntArray(Object... args) {
        super(args);
        value = new int[args.length - 1];
        for(int i = 1; i < args.length; i++) {
            value[i - 1] = (Integer) args[i];
        }
    }

    public AlleleIntArray(NBTTagCompound compound) {
        super(compound);
        readFromNBT(compound);
    }

    @Override
    public void readFromNBT(NBTTagCompound compound) {
        super.readFromNBT(compound);
        value = compound.getIntArray("value");
    }

    @Override
    public void writeToNBT(NBTTagCompound compound) {
        super.writeToNBT(compound);
        compound.setIntArray("value", value);
    }

}

File: src/main/java/seremis/geninfusion/soul/allele/AlleleIntArray.scala
package seremis.geninfusion.soul.allele

import net.minecraft.nbt.NBTTagCompound
import seremis.geninfusion.api.soul.EnumAlleleType
import seremis.geninfusion.soul.Allele

class AlleleIntArray(isDominant: Boolean, var value: Array[Int]) extends Allele(isDominant, EnumAlleleType.INT_ARRAY) {

    def this(args: AnyRef*) {
        this(args(0).asInstanceOf[Boolean], args.drop(1).asInstanceOf[Array[Int]])
    }

    def this(compound: NBTTagCompound) {
        super(compound)
        readFromNBT(compound)
    }

    override def readFromNBT(compound: NBTTagCompound) {
        super.readFromNBT(compound)
        value = compound.getIntArray("value")
    }

    override def writeToNBT(compound: NBTTagCompound) {
        super.writeToNBT(compound)
        compound.setIntArray("value", value)
    }
}
